waku-bindings/src/node/events.rs

```diff
@@ -37,23 +45,39 @@ pub struct WakuMessageEvent {
     pub waku_message: WakuMessage,
 }
 
-/// Register callback to act as event handler and receive application events,
-/// which are used to react to asynchronous events in Waku
-pub fn waku_set_event_callback<F: FnMut(Event) + Send + Sync>(ctx: &WakuNodeContext, mut f: F) {
-    let cb = |response: LibwakuResponse| {
-        if let LibwakuResponse::Success(v) = response {
-            let data: Event =
-                serde_json::from_str(v.unwrap().as_str()).expect("Parsing event to succeed");
-            f(data);
-        };
-    };
-
-    unsafe {
-        let mut closure = cb;
-        let cb = get_trampoline(&closure);
+impl WakuNodeContext {
+    pub fn new(obj_ptr: *mut c_void) -> Self {
+        Self {
+            obj_ptr: obj_ptr,
+            msg_observer: Arc::new(Mutex::new(Box::new(|_response| {
+                println!("msg observer not set")
```

Comment:
I wonder if instead of printing to stdout, we should trigger an error here, or maybe even change the type of `msg_observer` to `Option<Box<dyn FnMut(LibwakuResponse) + Send + Sync>>`, setting it to `None` by default, and in the nwaku side we'd determine whether the closure is nil or not and if nil, to log a warning indicating that there's no event callback set.